types/part_set.go

```diff
@@ -31,6 +32,11 @@ func (part *Part) ValidateBasic() error {
 	if len(part.Bytes) > int(BlockPartSizeBytes) {
 		return fmt.Errorf("too big: %d bytes, max: %d", len(part.Bytes), BlockPartSizeBytes)
 	}
+	// All parts except the last one should have the same constant size.
+	if part.Proof.Index < part.Proof.Total-1 &&
```

Comment:
Shouldn't this condition be using the actual Index of the part, rather than the index of the proof?